minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java

```diff
@@ -546,6 +550,37 @@ protected static void addConnection(final Element parentElement, ConnectionSchem
         }
     }
 
+    protected static String getConnectionName(ConnectionSchema connectionProperties) {
+        String connectionName = connectionProperties.getName();
+        if (StringUtil.isNullOrEmpty(connectionName)) {
+            return EMPTY_NAME;
+        }
+        return connectionName;
+    }
+
+    /**
+     * Will replace all characters not in [A-Za-z0-9_] with _
+     *
+     * This has potential for collisions so it will also append numbers as necessary to prevent that
+     *
+     * @param ids id map of already incremented numbers
+     * @param name the name
+     * @return a unique filesystem-friendly id
+     */
+    protected static String getUniqueId(Map<String, Integer> ids, String name) {
```

Comment:
I see what is happening here but wonder if hashing might be a better alternative.  While this does avoid collisions that could result, could keep things simpler given the requirement for unique names currently.